### PR TITLE
Major refactoring and CI

### PR DESCRIPTION
**Pull request by kostyrev**
I'd like to propose to get rid of version file.
Do you use this file for something else other than fot idempotency?
to review please use https://github.com/inkatze/wildfly/pull/15/files?w=1

### CODE REVIEW

**Comment by inkatze** [comment]
The version file has two purposes: avoid reinstalling the same version unless said otherwise (which apparently I forgot to add to this role) and idempotency which will be useful once I figure out how to add CI to the project.

I could merge your PR since the existence of the version file is pointless right now, unless you value idempotency, but I'd like to now if it'd make more sense if I added it again later on for the reasons listed above.

**Comment by kostyrev** [comment]
With this PR you avoid reinstalling the same version unless you delete manually /opt/wildfly-version dir
In the future we can add a task that deletes {{ wildfly_install_dir }} when some toggle is true.
This PR provides the idempotency.
I can send you a PR with [molecule](https://molecule.readthedocs.io) and Travis CI if you are interested.
See [this role](https://github.com/kostyrevaa/ansible-role-epel) for example

**Comment by inkatze** [comment]
I see. I would really appreciate a PR with molecule and Travis CI. And about this one, I think I can test it and merge it in a little while.

Thanks.

File: playbook.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+- hosts: all
```
Comment (inkatze) [inline playbook.yml]: Is it ok for the playbook not to have `become: yes`?

**Comment by inkatze** [comment]
Quick update: I still have to dig a little bit more into the refactoring and test it with the init service manager but is looking good so far.

File: playbook.yml (hunk 1 above)
Comment (kostyrev) [reply playbook.yml]: It's not necessary.
BTW, I fixed molecule and Travis issue.
See [this](https://travis-ci.org/kostyrevaa/wildfly/builds/180369421) successful Travis build.

**Comment by kostyrev** [comment]
I used initd as well.

**Comment by kostyrev** [comment]
Yet here are some other changes. I rebased my changes with latest changes in the role.
See this [build log](https://travis-ci.org/kostyrevaa/wildfly/builds/180401958)

**Comment by inkatze** [comment]
Everything looks great! I see you took care of a lot of use cases I never considered. 

Just one small thing though, I think we should stick with one format for boolean values. By the time I created this role the most widespread format was True/False. I think Yes/No looks nicer in YAML, but I don't have any particular preference for either and I'm not sure of what's the preferred format in the community. What do you think?

**Review by inkatze — APPROVED** [review]

**Comment by inkatze** [comment]
Created #20 to deal with the boolean situation.

**Comment by inkatze** [comment]
Fixes #17

**Comment by inkatze** [comment]
Resolves #17

**Comment by inkatze** [comment]
Thanks for your great contribution! Having CI will help a lot in the future and users will be happy for the improved running time.

**Comment by kostyrev** [comment]
About CI
You need to configure CI for your github account.
Now the Travis build badge (in README.md) points to my account.
Just follow [getting started](https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI%3A)

**Comment by inkatze** [comment]
I was so excited watching the builds working that I completely forgot about the badge.

---
Juan Diego Romero González

**Pull request merged**